### PR TITLE
feat: Warn user if FFmpeg is not installed

### PR DESCRIPTION
**Pull request by seakrueger**
Creates a Warning dialog on startup if the program cannot find FFmpeg orFFprobe in the PATH. Other interactions with the program are blocked until the issue is either ignored or resolved.

Uses native OS dialog, so should match Mac and Windows design paradigms on their respective systems (testers required to confirm)
image

### CODE REVIEW

File: tagstudio/src/qt/modals/ffmpeg_checker.py
Comment (eivl) [inline tagstudio/src/qt/modals/ffmpeg_checker.py:48]: Is there a logical or stylistic reason for checking negatives? wouldn't it be better to have it called `self.ffmpeg` and have it to `False` until you find it in `PATH` and set it to True, chance not needing to do any inversion in the return statement?

I know that I am sharing my own style here, and different styles are fine, I have just not seen this to many times so hence my question.

File: tagstudio/src/qt/modals/ffmpeg_checker.py
Comment (eivl) [inline tagstudio/src/qt/modals/ffmpeg_checker.py:41]: I dont like this implementation of which, it could just have been a method in your class, but since it ships with ffmpegs install it is fine to use. But, it uses `os.name` for platform checking, and I have seen many discussion about this usage and how it is better to use `platform.system()` to get the current platform name. 

It also uses os.path instead of `pathlib` 

I would like @CyanVoxel to comment on this. I can write the method if that is wanted as a new PR after this is merged. there is currently nothing wrong about this usage.

**Review by eivl — APPROVED** [review]
Approved once comments are resolved. 
Nice change!

File: tagstudio/src/qt/modals/ffmpeg_checker.py
Comment (seakrueger) [reply tagstudio/src/qt/modals/ffmpeg_checker.py]: No, I'm not sure why I flip-floppde the logic there, I had it the other way around previously. 🙃 Fixed -sorry

**Comment by seakrueger** [comment]
It may better to hold off until #460 is merged and update the url to point to the new docs site

**Comment by Nginearing** [comment]
> It may better to hold off until #460 is merged and update the url to point to the new docs site

As soon as #460 is merged, CyanVoxel plans to use the domain "tagstud.io" that is already bought, so the url will actually be something like docs.tagstud.io but it will still be github pages under the hood.

**Comment by CyanVoxel** [comment]
@seakrueger With #460 having been merged, the docs can be updated here 👍

**Comment by seakrueger** [comment]
The url now points to "https://docs.tagstud.io/help/ffmpeg/". Though it seems the new doc page will need to be cherry-picked to main for the url to work

**Comment by CyanVoxel** [comment]
The `doc` directory was also renamed to `docs` on main, just so you don't miss it

**Comment by seakrueger** [comment]
The actual docs page has now been spun off into its own PR (#473). The actual guide does not need to be part of this PR, and that solves the renamed dir and need to be on a different branch problems.

File: tagstudio/src/qt/modals/ffmpeg_checker.py
Comment (CyanVoxel) [reply tagstudio/src/qt/modals/ffmpeg_checker.py:40]: Could `shutil.which()` be used here instead?

Comment (seakrueger) [reply tagstudio/src/qt/modals/ffmpeg_checker.py:39]: I did not know the Python stdlib had a built in which command, using that instead.

**Comment by NeedNot** [comment]
Maybe just me but on Macos it says it can't find ffmpeg but the ffmpeg command works fine along with which or where ffmpeg

**Pull request merged**
